js/models/search/SearchProvider.js

```diff
@@ -0,0 +1,60 @@
+import app from '../../app';
+import is from 'is_js';
+import LocalStorageSync from '../../utils/backboneLocalStorage';
+import BaseModel from '../BaseModel';
+
+export default class extends BaseModel {
+  defaults() {
+    return {
+      name: '',
+      logo: '',
+      search: '', // currently not used, this searches vendors and listings
+      listings: '',
+      torsearch: '', // currently not used, this searches vendors and listings
+      torlistings: '',
+      locked: false,
+      order: 999999999, // order new providers after the defaults
+    };
+  }
+
+  localStorage() {
+    return new LocalStorageSync('__searchProviders');
+  }
+
+  sync(...args) {
+    return LocalStorageSync.sync.apply(this, args);
+  }
+
+  validate(attrs) {
+    const errObj = {};
+    const addError = (fieldName, error) => {
+      errObj[fieldName] = errObj[fieldName] || [];
+      errObj[fieldName].push(error);
+    };
+
+    if (attrs.name && is.not.string(attrs.name)) {
+      addError('name', app.polyglot.t('searchProviderModelErrors.name'));
+    }
+
+    if (attrs.logo && is.not.url(attrs.logo)) {
+      addError('name', app.polyglot.t('searchProviderModelErrors.logo'));
+    }
+
+    const urls = ['listings', 'torlistings'];
+    const validUrls = [];
+
+    urls.forEach((url) => {
+      if (attrs[url] && is.string(attrs[url]) && is.url(attrs[url])) {
+        validUrls.push(url);
+      }
+    });
+
+    if (!validUrls.length) {
```

Comment:
Why would you push a `listings` error if the torListing url is invalid? The error key should map directly to the field name.

```
    urls.forEach((url) => {
      if (attrs[url] && is.string(attrs[url]) && is.url(attrs[url])) {
         addError(url, app.polyglot.t(`searchProviderModelErrors.${url}`));
      }
    });
```

Reply:
The reason is it's an old copy and paste I never changed. Good catch!

Reply:
Ah, not a copy and paste. Listings is the name I was using for the listings search URL, which is confusing.

Showing a generic error if any of the urls are invalid isn't a great way to do this. I've rewritten it to show specific errors for each url, and then leave it up to the template to display the one that is relevant.